hal/common/retarget.cpp

```diff
@@ -707,3 +719,34 @@ extern "C" void __env_unlock( struct _reent *_r )
 #endif
 
 } // namespace mbed
+
+void* operator new  ( std::size_t count )
```

Comment:
Instead of overloading the mew operator, it is possible to set an handler which can intercept any allocation faillures.
More in formations here: http://en.cppreference.com/w/cpp/memory/new/set_new_handler